counterpartylib/lib/util.py

```diff
@@ -198,6 +202,82 @@ def get_asset_name (db, asset_id, block_index):
     elif not assets:
         return 0    # Strange, I know…
 
+# checks and validates subassets (PARENT.SUBASSET)
+#   throws exceptions for assset or subasset names with invalid syntax
+#   returns (None, None) if the asset is not a subasset name
+def parse_subasset_from_asset_name(asset):
+    subasset_parent = None
+    subasset_child = None
+    subasset_longname = None
+    chunks = asset.split('.', 1)
+    if (len(chunks) == 2):
+        subasset_parent = chunks[0]
+        subasset_child = chunks[1]
+        subasset_longname = asset
+
+        # validate parent asset
+        validate_subasset_parent_name(subasset_parent)
+
+        # validate child asset
+        validate_subasset_longname(subasset_longname, subasset_child)
+
+    return (subasset_parent, subasset_longname)
+
+# takes an asset description in 
+#   if it has a suffix like ;;lPARENT.child, then parse the subasset information
+#   throws exceptions for invalid asset names
+def parse_subasset_from_description(description):
+    subasset_parent = None
+    subasset = None
+    chunks = description.rsplit(';;l', 1)
```

Comment:
might want to add a comment that this function will get the right-most ';;l' and that additional ';;l' characters are ignored